File: src/domain/coral/agents.ts
import type { CoralAgentManifest } from '../../types'
import { listCoralAgentsNative, native } from '../../desktop/transport'

// Browser-dev fallback registry. Native desktop mode should ask Rust for the
// registry so the backend and UI agree on active Coral identities.
export const fallbackCoralAgents: CoralAgentManifest[] = [
  {
    id: 'worldcup-buyer-agent',
    displayName: 'World Cup Buyer',
    coralRole: 'buyer',
    service: 'txline',
    manifestPath: 'coral-agents/worldcup-buyer-agent/coral-agent.toml',
    description: 'Turns TxLINE triggers into WANTs, collects bids, awards the best seller, and starts policy-gated settlement.'
  },
  {
    id: 'seller-worldcup-edge',
    displayName: 'World Cup Edge Seller',
    coralRole: 'seller',
    service: 'txline.edge',
    manifestPath: 'coral-agents/seller-worldcup-edge/coral-agent.toml',
    description: 'Bids on odds movement WANTs and delivers a fixture-bound fair-line read.'
  },
  {
    id: 'seller-risk-policy',
    displayName: 'Risk Policy Seller',
    coralRole: 'seller',
    service: 'risk.policy',
    manifestPath: 'coral-agents/seller-risk-policy/coral-agent.toml',
    description: 'Prices downside, caps exposure, and outputs no-action/observe/simulate decisions.'
  },
  {
    id: 'seller-fan-card',
    displayName: 'Fan Card Seller',
    coralRole: 'seller',
    service: 'fan.card',
    manifestPath: 'coral-agents/seller-fan-card/coral-agent.toml',
    description: 'Converts match events into shareable fan-facing explanations.'
  },
  {
    id: 'verifier-agent',
    displayName: 'Verifier',
    coralRole: 'verifier',
    service: 'delivery.verify',
    manifestPath: 'coral-agents/verifier-agent/coral-agent.toml',
    description: 'Checks delivery hash, fixture binding, proof structure, and policy gates before release.'
  },
  {
    id: 'settlement-arbiter-agent',
    displayName: 'Settlement Arbiter',
    coralRole: 'settlement',
    service: 'settlement.release',
    manifestPath: 'coral-agents/settlement-arbiter-agent/coral-agent.toml',
    description: 'Bridges a verified run to the CoralOS settlement sidecar and devnet escrow observation.'
  }
]

// Load agent metadata from the strongest available source. If native IPC fails,
// the UI remains usable with the mirrored fallback list.
export async function loadCoralAgents(): Promise<CoralAgentManifest[]> {
  if (!native) return fallbackCoralAgents
  return listCoralAgentsNative().catch(() => fallbackCoralAgents)
}

File: src-tauri/src/coral/agents.rs
//! Coral agent registry exposed to the webview.
//!
//! The TOML manifests under `coral-agents/` are the intended future source of
//! truth. This built-in registry keeps the app working until manifest parsing is
//! promoted into runtime behavior.

use serde::Serialize;

// Public manifest summary returned by the `list_coral_agents` Tauri command.
#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CoralAgentManifest {
    pub id: &'static str,
    pub display_name: &'static str,
    pub coral_role: &'static str,
    pub service: &'static str,
    pub manifest_path: &'static str,
    pub description: &'static str,
}

// Return the agent identities used by the current market engine.
pub fn built_in_agents() -> Vec<CoralAgentManifest> {
    vec![
        CoralAgentManifest {
            id: "worldcup-buyer-agent",
            display_name: "World Cup Buyer",
            coral_role: "buyer",
            service: "txline",
            manifest_path: "coral-agents/worldcup-buyer-agent/coral-agent.toml",
            description: "Turns TxLINE triggers into WANTs, collects bids, awards the best seller, and starts policy-gated settlement.",
        },
        CoralAgentManifest {
            id: "seller-worldcup-edge",
            display_name: "World Cup Edge Seller",
            coral_role: "seller",
            service: "txline.edge",
            manifest_path: "coral-agents/seller-worldcup-edge/coral-agent.toml",
            description: "Bids on odds movement WANTs and delivers a fixture-bound fair-line read.",
        },
        CoralAgentManifest {
            id: "seller-risk-policy",
            display_name: "Risk Policy Seller",
            coral_role: "seller",
            service: "risk.policy",
            manifest_path: "coral-agents/seller-risk-policy/coral-agent.toml",
            description: "Prices downside, caps exposure, and outputs no-action/observe/simulate decisions.",
        },
        CoralAgentManifest {
            id: "seller-fan-card",
            display_name: "Fan Card Seller",
            coral_role: "seller",
            service: "fan.card",
            manifest_path: "coral-agents/seller-fan-card/coral-agent.toml",
            description: "Converts match events into shareable fan-facing explanations.",
        },
        CoralAgentManifest {
            id: "verifier-agent",
            display_name: "Verifier",
            coral_role: "verifier",
            service: "delivery.verify",
            manifest_path: "coral-agents/verifier-agent/coral-agent.toml",
            description: "Checks delivery hash, fixture binding, proof structure, and policy gates before release.",
        },
        CoralAgentManifest {
            id: "settlement-arbiter-agent",
            display_name: "Settlement Arbiter",
            coral_role: "settlement",
            service: "settlement.release",
            manifest_path: "coral-agents/settlement-arbiter-agent/coral-agent.toml",
            description: "Bridges a verified run to the CoralOS settlement sidecar and devnet escrow observation.",
        },
    ]
}
